### ISSUE TITLE
Segfault when installing via 'sudo make install'

### ISSUE THREAD

**Issue by DrTobe** [lxde/pcmanfm-qt#290]
I have updated my lxqt code today. Now, libfm-qt and pcmanfm-qt are two seperate projects, so I have built and installed libfm-qt first and then pcmanfm-qt. Trying to start pcmanfm-qt resulted in a segmentation fault. 

I have played around a little bit and found out that I can run the binary *pcmanfm-qt/build/pcmanfm/pcmanfm-qt* without a segmentation fault. I can also copy this binary to */usr/local/bin/* and run it then (which fixes my problem). But what is the difference between copying the binary to */usr/local/bin/* and running `sudo make install` ??

**Comment by pmattern** [comment]
This sounds like some incompatibility between old and new split versions.
Unfortunately you didn't tell which OS you're running and which way PCManFM-Qt and any eventual other LXQt components were installed before. Did you install those earlier versions from your distribution's packages? Would be good if you provided these information.

That said here are some suggestions how you may eventually track this further:

Checksum binary `pcmanfm-qt` in `/usr/bin`, `/usr/local/bin` and your build directory. If you installed PCManFM-Qt from a distribution's package figure out the checksum of the binary as provided by the package, too, e. g. by unpacking the package manually. Compare all those hashes so as to see which version in the FHS belongs to which source.
Have a look at environment variable PATH (`$ echo $PATH`).
Eventually the old binary is in a directory that's stated before the new binary in $PATH. That way the old binary would get used by default and could eventually cause the crash.

You may check the effect of `make install` by running `make DESTDIR=<path> install` were \<path\> is the *absoute* path of some arbitrary local directory and then run `find <path>`.
Eventually you will have to adjust the target paths that are chosen by `make install` by default. This can be done with cmake options like `-DCMAKE_INSTALL_PREFIX=/usr` or `-DCMAKE_INSTALL_LIBDIR=lib`, see e. g. section `build()` in `PKGBUILD` of [Arch Linux package pcmanfm-qt](https://projects.archlinux.org/svntogit/community.git/tree/trunk/PKGBUILD?h=packages/pcmanfm-qt).

**Comment by palinek** [comment]
Does the problem persist between reboots?

Note: the reason for such behaviour can also be the *ld.so* cache and linking **wrong** library in runtime. You can try to run *ldconfig* if it makes any difference.

**Comment by DrTobe** [comment]
Okay, I have found the reason but I do not know if it is my fault or a bug: The checksums of the two binaries were different (thanks for that hint) and I just guessed that the line
`-- Removed runtime path from "/usr/local/bin/pcmanfm-qt"`
looks suspicious.

I have found http://stackoverflow.com/questions/3352041/creating-binary-with-cmake-removes-runtime-path and the first (and accepted) answer solved the problem for me, too. I added `set_property(TARGET pcmanfm-qt PROPERTY INSTALL_RPATH_USE_LINK_PATH TRUE)` to *pcmanfm-qt/pcmanfm/CMakeLists.txt*.

Now, cmake prints out:
`-- Set runtime path of "/usr/local/bin/pcmanfm-qt" to "/home/myself/Qt/5.4/gcc_64/lib:/usr/local/lib"`
and I can start pcmanfm-qt after `make install`.

I do not know if the following information could be useful: I have installed Qt for development with the Qt installer in my home directory but there is another Qt version installed by the package manager because I additionally have KDE installed. Maybe, the pcmanfm-qt binary uses the wrong Qt libraries. Now, I am just wondering why I do not have any problems with the other lxqt applications ...

Having Qt installed twice was initially not intended but I had to do so because some lxqt components use private header files which weren't shipped with the Qt version from my package manager, if I remember right. I am running Ubuntu 15.10.
